### PR TITLE
refactor(docs): port i18n no-literal-string doctrine and catalogue reference (#1441)

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary

Implements #1441 per the merged implementation plan (`docs/specs/developments/20260811002516_1441-i18n-lint-rule-port/2_1441-i18n-lint-rule-port_implementation-plan.md`, Refactor path, no spec).

The plan's Template-Fit Check concluded the issue's literal ask ("port the lint rule with its test suite, and the catalogue/resolver skeleton") fails as stated: this template ships no application source tree and no JS/TS unit-test runner (Jest/Vitest) — the only tracked JS/TS surfaces are the placeholder `e2e/` Playwright scaffold and `hooks/` Haystack git-hook internals, neither a product application. The port therefore ships as documentation, not live code:

- **`docs/best-practices/stack/i18n.md`** (new): a worked-reference doc separating portable doctrine (no hard-coded user-facing strings; primary + fallback locale catalogues; enforcement must be machine-checked; planted-violation-proof discipline applied to the enforcement mechanism itself; data-driven-copy-through-catalogue pattern) from an illustrative React Native + i18next worked example (catalogue/resolver skeleton, ESLint config block, and a regex-based key-extraction scanner with its full E1-E13b edge-case table). Names the dynamic-key concatenation gap (`t('x.' + k)`) that a naive first version of the reference scanner accepted as a static key with zero dynamic findings — caught by a CodeRabbit review thread on the source PR before merge, closed with permanent E13/E13b regression tests.
- **`REVIEW.md`**: adds one conditional "Additional checks" block (Code Review Checklist, Pass 2, placed after the existing "automated check, guard, lint rule, or CI job" block) requiring, for projects that have adopted an i18n/catalogue convention, that the enforcement mechanism be active, catalogues stay in parity, and any new/modified enforcement mechanism carry a planted-violation proof per the existing generic block (delegates to it rather than restating it).
- **`docs/best-practices/3-testing.md`**: one-line cross-reference inside the existing "Planted-Violation Proofs" section pointing to `i18n.md`'s worked scanner/dynamic-key example.
- **`docs/best-practices/STACK-SPECIFIC.md`**: new "Internationalization (i18n)" row in the "Best Practices by Technology" table, mirroring the existing Supabase row.

## Reference material

Adapted from `lhpaul/personal-finances` issue #34 and merged PR [#41](https://github.com/lhpaul/personal-finances/pull/41):

- `apps/mobile/eslint.config.mjs` — the `i18next/no-literal-string` config block.
- `apps/mobile/src/i18n/{index,locale}.ts` — the runtime wiring and device-locale resolver.
- `apps/mobile/src/test-utils/catalogue-key-scan.ts` / `.test.ts` — the key-extraction scanner and its edge-case suite (E1-E12 in the original PR).
- Follow-up commit `306baae` ("fix(mobile): reject concatenated t() call keys in catalogue-key-scan") — the dynamic-key fix and its E13/E13b regression tests, applied as a same-PR follow-up responding to CodeRabbit thread `PRRT_kwDOTqQsJM6VvDJJ` on PR #41, before merge.
- PR #41's own three-run lint transcript (fails on a planted literal at `apps/mobile/src/dev/DesignSystemGallery.tsx:76:34`, passes before/after) and `grep -rn "no-literal-string" apps packages` showing zero `eslint-disable` usages — both cited in the new doc's "Source" section as the proof the worked example is production-proven.

## Verification

- `npx markdownlint-cli2` against all six affected/added Markdown files (the plan, `i18n.md`, `STACK-SPECIFIC.md`, `3-testing.md`, the smoke-test runbook, `REVIEW.md`): 0 issues.
- Heuristic lint (`scripts/lint/markdown-heuristic-lint.py`), scoped to `docs/specs/developments/20260811002516_1441-i18n-lint-rule-port/` and `docs/testing/workflow/`: 0 issues.
- `scripts/lint/check-changelog-duplicate-headers.sh CHANGELOG.md`: passed (no duplicate sub-headers, all section headings have link references).
- Smoke test runbook (`docs/testing/workflow/1441-i18n-lint-rule-port.smoke-test.md`) executed manually — all 5 steps and the assertions checklist confirmed against the shipped `i18n.md`/`REVIEW.md`/cross-reference content.

## Scope discipline

Implemented exactly what the merged plan specifies — no application code, no scripts, no CI changes, and no new automated check added to this repository. No files under `apps/`, `src/`, `scripts/lint/`, or any executable JS/TS path were added.

## Verification Discipline note (REVIEW.md)

This is a documentation-only PR that ships no executable check or lint rule in this repository. The literal planted-violation-proof obligation in `REVIEW.md`'s "PRs that add or modify an automated check, guard, lint rule, or CI job" block therefore does not bind this PR itself — no proof is fabricated here. The illustrative scanner code block inside `i18n.md` is not executed by this repository's own CI; its correctness is evidenced instead by the cited, already-merged, already-tested reference implementation (E1-E13b passing in `personal-finances`) and by this PR's own careful transcription, not by a fresh run in this repo.

## WHY_SAFE_TO_MERGE

- **scope**: Exactly the four files the merged plan specifies (`docs/best-practices/stack/i18n.md` new; `REVIEW.md`, `docs/best-practices/3-testing.md`, `docs/best-practices/STACK-SPECIFIC.md` each get one small, additive edit), plus the CHANGELOG entry. No application code, script, or CI file touched.
- **tests**: No unit/integration/CI test tier applies (documentation-only, per the plan's Testing Strategy). Verification is markdown lint (`markdownlint-cli2`, 0 issues), the heuristic lint script (0 issues), the CHANGELOG duplicate-header/link-reference check (passed), and the smoke-test runbook's 5 manual steps (all confirmed).
- **reviewer_outcome**: to be filled after Step 7/7a completes.
- **ci_outcome**: to be filled after Step 8 completes (head SHA: see below).
- **rollback_or_cleanup_risk**: Low. Pure documentation addition/edit; reverting the commit fully removes it with no data, schema, or runtime impact, and no other item in this batch touches the same files (per the plan's Cross-Cutting Operational Assumption Check).

## Summary by CodeRabbit

* **Documentation**
  * Added internationalization best practices covering translation catalogues, locale fallback, formatting, and data-driven copy.
  * Added a React Native example demonstrating locale resolution, runtime setup, catalogue usage, and translation checks.
  * Documented translation-key scanning limitations, dynamic keys, edge cases, and regression-testing guidance.
  * Updated technology references to include the internationalization guide.
  * Added review guidance for user-facing copy, locale parity, and enforcement changes.
  * Added a changelog entry summarizing the new internationalization guidance and review practices.

### CODE REVIEW

**Comment by lhpaul** [comment]
## Blocked: CI red on `ShellCheck static analysis` — pre-existing shared-tool bug, not this PR's content

Internal review (Step 7a) found this PR's content correct, plan-faithful, and
scope-clean (docs-only, exactly the four files + CHANGELOG the merged plan
specifies). However, CI is red:

```
$ python3 scripts/lint/workflow-shell-snippet-lint.py --base-ref origin/develop
docs/best-practices/stack/i18n.md:89: WS001: missing adjacent workflow-shell-contract marker (bash or bash-zsh)
docs/best-practices/stack/i18n.md:114: WS001: missing adjacent workflow-shell-contract marker (bash or bash-zsh)
docs/best-practices/stack/i18n.md:178: WS001: missing adjacent workflow-shell-contract marker (bash or bash-zsh)
```

**Root cause**: `workflow-shell-snippet-lint.py`'s `WS001` check treats any
fenced code block as "executable shell" whenever its content contains a
line starting with a `SHELL_SIGNAL` keyword (`export`, `if`, `while`, etc.)
— **regardless of the fence's declared language tag**. The three flagged
blocks are ` ```ts ` TypeScript reference code (the resolver skeleton and
key-extraction scanner), not Bash. This is not specific to this PR's
content: running the same script in `--all` mode shows the identical false
positive already present, unmodified, in `docs/best-practices/stack/supabase.md`'s
existing ` ```typescript ` blocks — it has simply never fired in CI because
CI only lints changed lines.

There is no in-repo way to silence this without either (a) rewriting the
TypeScript examples into unnaturally worded pseudo-code specifically to
dodge a scanner bug (degrades doc quality, doesn't match `supabase.md`'s
own established style), or (b) mislabeling the blocks with a
`workflow-shell-contract: bash` marker that misrepresents non-executable
TypeScript as executable Bash and pulls in the unrelated bash-quality
checklist (WS002/WS006).

**Not fixed inline**: the correct fix lives in
`scripts/lint/workflow-shell-snippet-lint.py`, outside this PR's approved
scope (the merged plan's Layer-by-Layer Changes explicitly lists "None — no
script or shell-helper changes" for Shared Packages/Libraries). The fix also
has repo-wide blast radius (`workflow-shell-snippet-lint.py`'s `ROOTS`
covers `docs/best-practices/`, `docs/workflow/`, skill/agent/command files
repo-wide) and more than one defensible design (exclude known non-shell
language tags vs. narrow `SHELL_SIGNAL`'s keyword set vs. add a "not
executable" contract value) — the kind of tradeoff `REVIEW.md`'s Fix vs.
Report guidance reserves for a human/separate PR rather than an inline fix
here.

**Filed as a separate tool-fix backlog item**: #1468 (Type: Workflow),
with the reproduction, the `supabase.md` cross-check proving this
pre-existing, and a proposed fix sketch.

**Status**: this PR (#1467) stops here per the `failing_ci` guardrail stop
condition — genuinely CI-red, and the fix is out of this item's scope. It
is not applied `ready-for-human-review` while CI is red. Recommended path:
merge #1468's fix first (or get explicit human approval to widen this PR's
scope to include the lint-script fix), then re-run CI on this branch.

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean — no blocking findings
**Platforms:** pr-agent (clean), coderabbit (clean)
**Findings:** 0 blocking, 0 suggestions

**Ready reviewer phase:** No net-new blocker was found after the draft GitHub gate.
**Ready-phase platforms:** coderabbit

**Advisory findings (non-blocking):**
- Possible Issue ([view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1467#issuecomment-5247971182))

*Posted automatically by `pr-review-loop.sh`.*

<details>
<summary>Reviewer-loop history (6 iterations)</summary>


```json
{
  "schema": "reviewer_loop_history.v1",
  "pr_number": 1467,
  "updated_at": "2026-08-11T16:44:35Z",
  "history_status": "available",
  "history_unavailable_reason": "",
  "entries": [
    {
      "iteration": 1,
      "recorded_at": "2026-08-11T01:34:11Z",
      "head_sha": "88d2ada13b07c2ad951d3395d19ce808822bca0a",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    },
    {
      "iteration": 2,
      "recorded_at": "2026-08-11T15:18:28Z",
      "head_sha": "b03d37e85bb5d77608309e01f9f8a959697655d2",
      "result": "needs_fixes",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (needs_fixes)"
      ],
      "blocking_count": 1,
      "suggestion_count": 3,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "coderabbit"
      }
    },
    {
      "iteration": 3,
      "recorded_at": "2026-08-11T16:17:00Z",
      "head_sha": "3ee6d00316b0b3120c18ea85dd565d3403e1914f",
      "result": "needs_fixes",
      "reason": "coderabbit_unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (needs_fixes)"
      ],
      "blocking_count": 3,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "coderabbit"
      }
    },
    {
      "iteration": 4,
      "recorded_at": "2026-08-11T16:27:24Z",
      "head_sha": "1286205dddfc0cdebd7aa8078d4907402f5296b3",
      "result": "needs_fixes",
      "reason": "coderabbit_unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (needs_fixes)"
      ],
      "blocking_count": 1,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "coderabbit"
      }
    },
    {
      "iteration": 5,
      "recorded_at": "2026-08-11T16:29:17Z",
      "head_sha": "1286205dddfc0cdebd7aa8078d4907402f5296b3",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    },
    {
      "iteration": 6,
      "recorded_at": "2026-08-11T16:44:35Z",
      "head_sha": "d96e70c0f4d19c82cbc53cb77cf6d52c9ca9d739",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    }
  ]
}
```
</details>

**Comment by lhpaul** [comment]
## Smoke Test Results

Ran `docs/testing/workflow/1441-i18n-lint-rule-port.smoke-test.md` manually against the shipped diff.

- **Step 1 (new reference doc)**: `docs/best-practices/stack/i18n.md` contains the Doctrine, "Worked example: React Native + i18next" (catalogue shape, resolver, ESLint config), "Key-extraction scanner and the dynamic-key pitfall" (with edge-case table), and "What is portable / not portable" sections. The worked-example section opens with an explicit "one illustrative stack, not a universal requirement" callout. PASS.
- **Step 2 (dynamic-key pitfall documented by name)**: edge-case table includes both static cases (E1, E9) and the dynamic concatenation cases E13 (`t('a.x' + suffix)`) / E13b (`t('a.' + section + '.title')`), and the dedicated "The dynamic-key pitfall" subsection names the naive-scanner regression, the CodeRabbit thread that caught it, and the concrete fix (require an argument terminator immediately after the closing quote). PASS.
- **Step 3 (REVIEW.md checklist addition)**: new "Additional checks for **PRs that add or modify user-facing copy in a project with a configured i18n / catalogue convention**" block added to Code Review Checklist Pass 2, immediately after the existing "automated check, guard, lint rule, or CI job" block. Includes an explicit applicability gate, checks enforcement-is-active and catalogue-parity, and delegates (does not restate) the planted-violation-proof requirement to the existing generic block. PASS.
- **Step 4 (cross-references)**: `docs/best-practices/STACK-SPECIFIC.md` has a new "Internationalization (i18n)" row linking to `stack/i18n.md`, mirroring the Supabase row. `docs/best-practices/3-testing.md`'s "Planted-Violation Proofs" section has a one-line pointer to `i18n.md`'s scanner/dynamic-key section. Neither duplicates `i18n.md`'s content. PASS.
- **Step 5 (markdown lint)**: `npx markdownlint-cli2` against all six files (plan, `i18n.md`, `STACK-SPECIFIC.md`, `3-testing.md`, smoke-test runbook, `REVIEW.md`) — 0 issues. Heuristic lint (`scripts/lint/markdown-heuristic-lint.py`, scoped to the plan's directory and `docs/testing/workflow/`) — 0 issues (exit 0). PASS.
- **Assertions checklist**: all six boxes confirmed, including "No files under `apps/`, `src/`, `scripts/lint/`, or any executable JS/TS path were added" (verified via `git diff --stat` on this PR's branch — only the five documentation files + CHANGELOG changed).

**Overall**: PASS. No defects found by the smoke test itself; the PR is currently blocked on an unrelated CI infrastructure issue (see the other comment on this PR / issue #1468), not on smoke-test failures.

**Comment by lhpaul** [comment]
@coderabbitai resume

**Comment by lhpaul** [comment]
@coderabbitai review

**Comment by lhpaul** [comment]
@coderabbitai resume

**Comment by lhpaul** [comment]
@coderabbitai resume

**Comment by lhpaul** [comment]
@coderabbitai review

File: docs/best-practices/stack/i18n.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+import { resolveDeviceLocale } from './locale';
+
+void i18n.use(initReactI18next).init({
+  resources: { es: { translation: es }, en: { translation: en } },
+  lng: resolveDeviceLocale(),
+  fallbackLng: 'es',
+  keySeparator: false,
+  interpolation: { escapeValue: false }, // the UI layer already escapes values
+  react: { useSuspense: false }, // resources are bundled synchronously above
+});
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/best-practices/stack/i18n.md:143]: Fixed in eec1633: added `initImmediate: false` (with an explanatory comment) to the `i18n.use(initReactI18next).init({...})` call in this example, pinning the synchronous-resources + `useSuspense: false` combination so it never observes a still-initializing instance. See the updated code block above.

**Comment by lhpaul** [comment]
@coderabbitai review

**Comment by lhpaul** [comment]
## /run-epic PR Disposition

- Scope source: resume dispatch via /run-items (Protocol 91 explicit_list batch) — issue #1441 refactor item; blocker issue #1468 (WS001 lint false positive) resolved and merged upstream (PR #1469) prior to this resume
- Item: #1441 - Port the no-literal-string i18n lint rule and catalogue skeleton from personal-finances
- PR: #1467
- Reviewed head SHA: `d96e70c0f4d19c82cbc53cb77cf6d52c9ca9d739`
- Reviewer result: clean
- Blocking findings: 0
- Advisory findings: 1
- Risk: low
- Merge authority: delegated mode with may_merge_pr: true and max_merge_risk: medium at the guardrails level, but THIS RUN's explicit dispatch instructions reserved merge authority for the parent/orchestrator (Gate 5 + scoped Protocol 94 merge) — this runner stopped at ready-for-human-review without merging
- Final decision: ready_for_human_review

### Risk Reasons

- Docs-only change (docs/best-practices/stack/i18n.md new; small additive edits to REVIEW.md, docs/best-practices/3-testing.md, docs/best-practices/STACK-SPECIFIC.md; CHANGELOG entry)
- No application source, script, or CI-behavior change in this PR's own diff (the merge commit that resumed this PR only synced in an already-merged, already-reviewed lint-script fix from develop; it added zero net lines to this PR's diff vs. develop)
- Four review rounds with CodeRabbit and PR-Agent, all findings resolved or independently fact-checked (one PR-Agent claim was verified inaccurate via a live Node repro before writing the correction, to avoid propagating a wrong root-cause into the doc)
- All CI checks green (Markdown lint, ShellCheck, PR-Agent, policy; E2E regression skipped as configured placeholder)

### Why Safe to Merge

- Scope: PR now contains both the original #1441 documentation-only work (new docs/best-practices/stack/i18n.md; small additive edits to REVIEW.md, docs/best-practices/3-testing.md, docs/best-practices/STACK-SPECIFIC.md; CHANGELOG) and a merge commit from develop carrying #1469's lint-script fix (scripts/lint/workflow-shell-snippet-lint.py + its test) plus that PR's own CHANGELOG entry. The merge commit added zero new lines to this PR's own diff against develop (verified via `git diff origin/develop...HEAD --stat`, unchanged before/after the merge) — it only resolved a divergence with develop, it did not introduce new content requiring re-review. All fixer commits after the merge (5 total, across 3 CodeRabbit/PR-Agent review rounds) stayed within docs/best-practices/stack/i18n.md, docs/best-practices/3-testing.md, and REVIEW.md — no expansion into scripts/lint/ or any other file outside the merged plan's scope.
- Tests: No executable test suite applies — this is a documentation-only PR shipping reference/illustrative material for downstream projects (per the merged plan's Template-Fit Assessment, this template repo has no application source tree or unit-test runner to host the ported React Native/i18next code). Verification performed instead: markdownlint-cli2 (0 issues, all touched files, each round), scripts/lint/markdown-heuristic-lint.py (0 issues), scripts/lint/check-changelog-duplicate-headers.sh (pass), scripts/lint/workflow-shell-snippet-lint.py --base-ref origin/develop (exit 0 — the original CI blocker), the smoke test runbook at docs/testing/workflow/1441-i18n-lint-rule-port.smoke-test.md (PASS, all 5 steps, executed in the prior run), and one PR-Agent finding (escaped-quote handling in the illustrative key-extraction scanner) that was independently verified against actual behavior via a live Node.js repro before writing the corrective doc text — the finding's literal claim (returns null / dynamic) was found inaccurate on inspection; the real, narrower defect (retains the raw backslash instead of unescaping) was documented instead.
- Reviewer outcome: pr-review-loop.sh RESULT=clean on the final head (d96e70c), 0 blocking, 0 suggestions, both pr-agent and coderabbit clean, ready-phase (coderabbit) clean with no net-new blocker, post-clean recheck found no late-arriving threads. Independently re-verified via direct GraphQL reviewThreads query (0 unresolved) and the Step 8a pre-Check-4 bot-authored-unresolved-thread filter (0). Across the full resume, 3 review rounds surfaced 1 Major + 7 Minor/advisory findings total; all were fixed in-file (docs-only) except one that was replied-to and manually resolved via the GraphQL resolveReviewThread mutation after its underlying content was already fixed in an earlier commit but GitHub had not auto-resolved the stale thread. One remaining PR-Agent advisory (escaped-quote scanner edge case, E14) was fixed by adding a documented edge-case row, independently fact-checked against the finding's literal claim before writing the fix.
- CI outcome: pr-ci-loop.sh RESULT=green on final head d96e70c0f4d19c82cbc53cb77cf6d52c9ca9d739 — 7 total checks, 0 failing, 0 pending. Direct check-runs query confirms: Markdown lint (spec/plan/CHANGELOG) success, ShellCheck static analysis success (the original CI blocker, now resolved via #1469's merged fix), PR-Agent review success, policy success, E2E regression (placeholder) skipped as configured.
- Rollback or cleanup risk: Very low. Pure documentation addition/edit with no runtime, build, script, or CI-behavior change in this PR's own diff — reverting is a simple git revert with no data, schema, config, or dependency implications. No new files outside docs/ and CHANGELOG.md. No downstream project consumes this doc programmatically (it is human-read reference material), so there is no cross-repo breakage risk from merging or reverting.

### Invocation Policy

Not recorded.

### Checkpoint Policy

Not recorded.

### Advisory Decisions

| Source | Category | Decision | Rationale |
| --- | --- | --- | --- |
| PR-Agent | Possible Issue — escaped-quote handling in illustrative key-extraction scanner (parseFirstArgument) | fixed | Independently verified via a live Node.js repro (not by trusting the finding's stated root cause) that the finding's literal claim was inaccurate: the escaped quote is correctly skipped past by the closing-quote scan, so the call is NOT reported as a dynamic finding as PR-Agent claimed. The real, narrower defect is that the extracted value retains the raw backslash (`it\'s`) instead of being unescaped (`it's`), which would produce a false missing-key finding against a real catalogue. Documented this accurately as a new edge-case row (E14) in the scanner's edge-case table, following the same convention already established for E10/E10b in this PR. |

### Verification Evidence

- Labels: ready-for-human-review, ready-for-regression, Review effort 2/5
- CI result: green
- Reviewer summary present: true
- Unresolved thread count: 0
- PR merge state: MERGEABLE
- Issue state: open
- Project status: to be updated to Development in Review (Step 8b) immediately after this record is posted

### Protocol Deviations

| Action | Impact | Mitigation |
| --- | --- | --- |
| Manually replied to and resolved (via GraphQL resolveReviewThread mutation) a stale CodeRabbit review thread whose underlying content had already been fixed two commits earlier but which GitHub had not auto-marked resolved. | None — content was already correct; this was purely a GitHub thread-state bookkeeping action required to pass the Step 8a pre-Check-4 unresolved-thread gate honestly rather than relying on the reviewer loop's own (looser) content-based clean determination. | Posted an explicit reply on the thread citing the fixing commit SHA before resolving, so the audit trail is traceable from the GitHub UI alone. |
